### DISCUSSION TITLE
Segmentation fault (core dumped) on cuda 12.4 & cudnn 9.3.0

### DISCUSSION THREAD

**Discussion by mv2050arshad** [davisking/dlib#2994] [Q&A] [answered]
I build dlib (cuda-active) and i tried to build my application using dlib Face recognition. 
Everything build successfully but i got run-time error :
Segmentation fault (core dumped)

It was detected faces and gpu , but i think the problem was when it try to load model in gpu for landmark extraction.

cout<<  dlib::cuda::get_num_devices() ;   // print  1
 dlib::cuda::set_device(0); // without error run this also

Whats wrong? what is the last version of cuda/cudnn which supported by dlib? 

I installed : 
Cuda 12.4
Cudnn  9.3.0
Gpu 4070

**Comment by arrufat** [comment]
Could you try passing the following flag when building dlib? `-DDLIB_USE_CUDA_COMPUTE_CAPABILITIES=89`

**Reply by mv2050arshad** [reply]
Thanks for your answer,
okay i try to do that and will tell you the answer. Do you think that dlib support by that cuda and cudnn version ? how much the maximum version that supported by dlib ? i have to use Ubuntu 22.04, i can not to install some lower version.

**Reply by arrufat** [reply]
I always have the latest versions of everything (I use Arch btw) and it works. Maybe that GPU/CUDA version does not work with compute capabilities 50? I don't know. I always check what are the compute capabilities of the GPU at hand and enable those when building dlib.

**Reply by mv2050arshad** [reply]
I have changed it to 89 and build dlib sucessfully. But when i use it in my code, i have compiled error now : 
I changed cuda to 11.7 and cudnn to 8.5 also. Because i thought maybe new version not supported by dlib.
![2er](https://github.com/user-attachments/assets/157ab80d-3a97-4879-bfb9-10bdb8609022)

Its seems some problem in link library but i linked it i think : 

unix:!macx: LIBS += -L$$PWD/../../../../usr/local/cuda/lib64/ -lcudnn  -lcudart  -lcublas  -lcurand -lcusolver
INCLUDEPATH += $$PWD/../../../../usr/local/cuda/include
DEPENDPATH += $$PWD/../../../../usr/local/cuda/include

**Reply by arrufat** [reply]
Oh, so you're not using CMake to build your project?

**Reply by mv2050arshad** [reply]
I used qmake, is'nt work ?

**Reply by arrufat** [reply]
Then I'm not sure, CMake will set the flags for you, I don't know about qmake.

**Reply by mv2050arshad** [reply]
I build version 19.23 and it build with my application, but all the time got same error : Segmentation fault (core dumped).
i use ubuntu 22.04 and gcc 11.3, i changed  CUDA_COMPUTE_CAPABILITIES to 89, 86 , 80, 50. No one work and same error. 
i set_device(0) before make net, nothing change. Have you ever successful to use cuda on new versions of ubuntu or cuda or dlib?

**Reply by arrufat** [reply, marked as answer]
I always use dlib from master, mostly on Arch Linux. However, at work, I have one machine with Ubuntu and:
- nvidia drivers 560
- cuda 12.6
- cudnn 9.3

And everything works building with CMake.

**Reply by mv2050arshad** [reply]
Thank you so much for your answers, it means that there is a way and have to correct my bugs.
